### PR TITLE
Add sphinx docs

### PR DESCRIPTION
**Pull request by accraze**
This PR adds a Sphinx documentation site.

How to build docs site locally: (w/ python 3)

```
cd docs
make html
cd _build/html
python -m http.server
```
The site should be available at http://localhost:8000

https://phabricator.wikimedia.org/T230517

### CODE REVIEW

File: docs/requirements.txt [hunk 1]
```diff
@@ -0,0 +1,4 @@
+sphinx>=1.3
+sphinx-py3doc-enhanced-theme
+m2r
+-e .
```
Comment (halfak) [inline docs/requirements.txt]: I wonder if these should be in a special `docs-requirements.txt` file.

File: drafttopic/__init__.py [hunk 2]
```diff
@@ -1,3 +1,17 @@
...
+new drafts based on Wikiprojects on English Wikipedia.
+
+In this package, you'll find the
+:mod:`feature lists <drafttopic.feature_lists>` used to train models
+for the supported wiki.  There's a set of
+:mod:`command-line utilities <drafttopic.utilities>`
+that are used to perform data pipeline operations specific to training and
+testing draft t opic models.
+
+See the :ref:`API Reference <api_reference>` for low level details.
```
Comment (halfak) [inline drafttopic/__init__.py]: I don't think this `:ref:` actually links to anything.

File: docs/reference/drafttopic.utilities.tests.rst [hunk 3]
```diff
@@ -0,0 +1,54 @@
+drafttopic.utilities.tests package
```
Comment (halfak) [inline docs/reference/drafttopic.utilities.tests.rst]: We probably don't want the tests showing up in our docs.

File: docs/requirements.txt (hunk 1 above)
Comment (halfak) [reply docs/requirements.txt]: Aha!  I see this is in "docs/requirements.txt" which is OK with me.

Comment (accraze) [reply docs/requirements.txt]: This is actually recommended for the readthedocs build process: https://docs.readthedocs.io/en/stable/guides/specifying-dependencies.html?highlight=requirements#specifying-a-requirements-file

File: docs/reference/drafttopic.utilities.tests.rst (hunk 3 above)
Comment (accraze) [reply docs/reference/drafttopic.utilities.tests.rst]: Agreed, will remove.

File: drafttopic/__init__.py (hunk 2 above)
Comment (accraze) [reply drafttopic/__init__.py]: fixed it!

**Pull request merged**